File: solutions/r/prime-factors/1/prime-factors.R
prime_factors <- function(number) {
  if(number <= 1) return(c())
  if(number <= 3) return(c(number))
  max <- floor_sqrt(number)
  index <- 1
  primes <- rep(NA,max)
  while(number > 1) {
    max <- floor_sqrt(number)
    light <- TRUE
    if(max > 1) {
      for(i in 2:max) {
        if(number%%i == 0) {
          primes[index] <- i
          index <- index + 1
          number = number/i
          light <- FALSE
          break
        }
      }
    }
    if(light) {
      primes[index] <- number
      break
    }
  }
  return(primes[!is.na(primes)])
}

floor_sqrt <- function(number) {
  for(i in 1:number) {
    if(i*i < number) {
      next
    } else if(i*i == number) {
      return(i)
    } else {
      return(i-1)
    }
  }
}

File: solutions/julia/prime-factors/1/prime-factors.jl
function prime_factors(n)
    primes = []
    prospect = 3
    while n > 1
        while n%2 == 0
            push!(primes,2)
            n = n/2
        end
        if n <= 1 return primes end
        if n%prospect == 0
            push!(primes,prospect)
            n = n/prospect
            prospect = 3
            continue
        end
        prospect += 2
    end
    return primes
end
